mozfldp/fed_avg_w_dp.py

```diff
@@ -0,0 +1,250 @@
+        )
+
+        theta = prev_theta + merged_user_values + rand_gauss_noise
+        prev_theta = theta
+
+        print("Theta ([coef, inter] for round {}: \n {}".format(round_t, theta))
+        _moments_accountant_accum_priv_spending(prms.noise_scale)
+
+    privacy_spent = _calc_privacy_spent()
+    print("Total privacy spent from privacy budget: {}".format(privacy_spent))
+
+    coef_slice, inter_slice = _get_coef_and_inter_slice_from_theta(
+        theta, prms.num_features, prms.num_labels
+    )
+
+    return np.array(coef_slice), np.array(inter_slice)
+
+
+# TODO: Give better function name...
+def _merge_all_user_weights(user_sel_prob, weight_sum, user_updates_buf, user_weights):
+    """
+    Merge all user updates for a round into a single delta (vector).
+    """
+
+    num_users_in_batch = len(user_updates_buf)
+    merged_weights = np.zeros(len(user_updates_buf[0]), dtype=np.float64, order="C")
+
+    for i in range(num_users_in_batch):
+        weighted_user_val = np.multiply(user_updates_buf[i], user_weights[i])
+        merged_weights = np.add(merged_weights, weighted_user_val)
+
+    merged_weights = np.divide(merged_weights, user_sel_prob * weight_sum)
+    return merged_weights
+
+
+def flat_clip(sensitivity, vec):
+    vec *= min(1, sensitivity / np.linalg.norm(vec))
+    return vec
+
+
+def user_update_fed_avg(prms, round_user_features, round_user_labels, theta_0):
+
+    batches_features = []
+    batches_labels = []
+
+    classifier = SGDClassifier(
+        loss="hinge", penalty="l2", max_iter=1, random_state=prms.rand_seed
+    )
+
+    round_num_entries = len(round_user_features)
+
+    # For now just skip any batches that are smaller than batch_size
+    num_batches = round_num_entries // prms.batch_size
+
+    theta = np.array(theta_0, copy=True)
+
+    # Split the round data into seperate batches
+    for i in range(0, round_num_entries, prms.batch_size):
+        batches_features.append(round_user_features[i : i + prms.batch_size])
+        batches_labels.append(round_user_labels[i : i + prms.batch_size])
+
+    for _ in range(prms.num_epochs):
+        for j in range(num_batches):
+
+            coef, inter = _get_coef_and_inter_slice_from_theta(
+                theta, prms.num_features, prms.num_labels
+            )
+
+            classifier.fit(
+                batches_features[j],
+                batches_labels[j],
+                coef_init=coef,
+                intercept_init=inter,
+            )
+
+            _set_coef_and_inter_on_theta(theta, classifier.coef_, classifier.intercept_)
+            theta = theta_0 + flat_clip(prms.sensitivity, theta - theta_0)
+
+    return theta - theta_0
+
+
+def _get_random_selection_of_user_idxs(num_users):
+    # For now just randomly pick 1/5th of the users.
+    # TODO: Change this later!
+    return np.random.choice(num_users, num_users // 5)
+
+
+def _init_theta_from_moment_accountant(num_features, num_labels):
+    # Just a placeholder value for now
```

Comment:
As mentioned above, drop the "moment accountant" reference from the function name.

Reply:
Done.